src/rx-query.ts

```diff
@@ -524,30 +632,26 @@ export function createRxQuery<RxDocType>(
 
     // ensure when created with same params, only one is created
     ret = tunnelQueryCache(ret);
+    // TODO: clear persistent query cache as well
     triggerCacheReplacement(collection);
 
     return ret;
 }
 
 /**
  * Check if the current results-state is in sync with the database
- * which means that no write event happened since the last run.
+ * which means that no writes event happened since the last run.
  * @return false if not which means it should re-execute
  */
 function _isResultsInSync(rxQuery: RxQueryBase<any>): boolean {
     const currentLatestEventNumber = rxQuery.asRxQuery.collection._changeEventBuffer.counter;
-    if (rxQuery._latestChangeEvent >= currentLatestEventNumber) {
-        return true;
-    } else {
-        return false;
-    }
```

Comment:
yeah IDK what's up with the original code, this is a pretty basic Programming 101 mistake